packages/react-dom/src/client/ReactInputSelection.js

```diff
@@ -13,7 +13,95 @@ import * as ReactDOMSelection from './ReactDOMSelection';
 import {ELEMENT_NODE} from '../shared/HTMLNodeType';
 
 function isInDocument(node) {
-  return containsNode(document.documentElement, node);
+  return (
+    node.ownerDocument && containsNode(node.ownerDocument.documentElement, node)
+  );
+}
+
+function getActiveElementDeep() {
+  var win = window;
+  var element = getActiveElement();
+  while (element instanceof win.HTMLIFrameElement) {
+    try {
+      win = element.contentDocument.defaultView;
+    } catch (e) {
+      return element;
+    }
```

Comment:
Why a try/catch? Could you just do:

```
if (!element.contentDocument) {
  return element
}
```
?

Reply:
If you just try to access the `contentDocument` of an `HTMLIframeElement`that comes from a non-marching separate domain (like if there’s an embedded YouTube player in the DOM), the browser will throw an error.